internal/distro/rhel9/imagetype.go

```diff
@@ -305,7 +288,7 @@ func (t *imageType) PackageSets(bp blueprint.Blueprint, options distro.ImageOpti
 	}
 
 	// create a manifest object and instantiate it with the computed packageSetChains
-	manifest, err := t.initializeManifest(&bp, options, globalRepos, packageSets, containers, 0)
+	manifest, err := t.initializeManifest(&bp, options, repos, packageSets, containers, 0)
```

Comment:
This is relevant for all distros: So from now on, the `packageSets` won't ever include any `Repositories` set when passed to `initializeManifest`, right?

If yes, then we should probably rethink how relevant are things like passing the package sets into specific pipelines such as https://github.com/osbuild/osbuild-composer/blob/96b42b57919440b798b2171f45792efcf1c148f1/internal/distro/rhel9/images.go#L492 then resulting in https://github.com/osbuild/osbuild-composer/blob/96b42b57919440b798b2171f45792efcf1c148f1/internal/image/image_installer.go#L77 and repositories (there should not be any) in https://github.com/osbuild/osbuild-composer/blob/96b42b57919440b798b2171f45792efcf1c148f1/internal/manifest/anaconda.go#L125-L136

If my assessment is correct, then the code in anaconda pipeline (and other that handle repos, such as OS pipeline) handling extra repos should become unnecessary with this change. IOW the `packageSets` should never contain any repos, `ExtraBaseRepos` in pipeline definitions should be probably removed and `packageSets` passed to `initializeManifest()` with repositories set should be probably considered a programming error.

But I'm not 100% sure about all of this... 🤔

Reply:
All good points.  I think you're right.  Need to dig a bit and clarify everything while fixing the payload repositories.

Reply:
This really broke my brain for a whole day.  See my [followup comment below](https://github.com/osbuild/osbuild-composer/pull/3291#issuecomment-1440960864).